### PR TITLE
Music fixes

### PR DESCRIPTION
**Pull request by samdivaio**

### CODE REVIEW

File: src/NadekoBot/Modules/Music/Classes/Song.cs [hunk 1]
```diff
@@ -46,7 +47,25 @@ public class Song
 
         public string PrettyFullTime => PrettyCurrentTime + " / " + PrettyTotalTime;
 
-        public string PrettyName => $"**[{SongInfo.Title.TrimTo(70)}]({SongInfo.Query})**";
+        //public string PrettyName => $"**[{SongInfo.Title.TrimTo(70)}]({SongInfo.Query})**";
```
Comment (Kwoth) [inline src/NadekoBot/Modules/Music/Classes/Song.cs]: Why this?

File: src/NadekoBot/Modules/Music/Classes/Song.cs [hunk 2]
```diff
@@ -67,16 +86,36 @@ public class Song
 
         public string Thumbnail {
             get {
-                switch (SongInfo.ProviderType)
+                switch (SongInfo.Provider)
```
Comment (Kwoth) [inline src/NadekoBot/Modules/Music/Classes/Song.cs]: what's wrong with this?

File: src/NadekoBot/Modules/Music/Music.cs [hunk 3]
```diff
@@ -221,12 +221,12 @@ public Task Pause(IUserMessage umsg)
...
                     .WithAuthor(eab => eab.WithName("Now Playing")
                         .WithMusicIcon())
-                        .WithTitle($"{currentSong.SongInfo.Title}")
+                        .WithTitle(currentSong.SongInfo.Title)
                     .WithDescription(currentSong.PrettyFullTime)
                     .WithFooter(ef => ef.WithText($"{currentSong.PrettyProvider} | {currentSong.QueuerName}"))
                     .WithOkColor()
                     .WithThumbnail(tn => tn.Url = currentSong.Thumbnail)
-                    .WithUrl(currentSong.SongInfo.Query);
+                    .WithUrl(currentSong.songURL);
```
Comment (Kwoth) [inline src/NadekoBot/Modules/Music/Music.cs]: Why? It worked just fine before?

Comment (samdivaio) [inline src/NadekoBot/Modules/Music/Music.cs]: If you use local song, the now playing will not work, its a fix for that.

File: src/NadekoBot/Modules/Music/Classes/Song.cs (hunk 2 above)
Comment (samdivaio) [inline src/NadekoBot/Modules/Music/Classes/Song.cs]: Soundcloud artwork can't get avatar using ProviderType, unless soundcloud url is used. Its a fix.

File: src/NadekoBot/Modules/Music/Classes/Song.cs (hunk 1 above)
Comment (samdivaio) [inline src/NadekoBot/Modules/Music/Classes/Song.cs]: Local songs gives out a name [blablabla](/root/music/etc) its a fix for that

Comment (samdivaio) [inline src/NadekoBot/Modules/Music/Classes/Song.cs]: ```[blablabla](/root/music/etc) ```

File: src/NadekoBot/Modules/Music/Classes/Song.cs (hunk 2 above)
Comment (samdivaio) [inline src/NadekoBot/Modules/Music/Classes/Song.cs]: *soundcloud can't get artwork

File: src/NadekoBot/Modules/Music/Music.cs (hunk 3 above)
Comment (samdivaio) [inline src/NadekoBot/Modules/Music/Music.cs]: Because of invalid url

File: src/NadekoBot/Modules/Music/Classes/Song.cs [hunk 4]
```diff
@@ -104,6 +143,7 @@ public Song Clone()
         {
             var s = new Song(SongInfo);
             s.MusicPlayer = MusicPlayer;
+			s.QueuerName = QueuerName;
```
Comment (samdivaio) [inline src/NadekoBot/Modules/Music/Classes/Song.cs]: This one is one of the fix for the queuer name being blank, mostly happen with goto

**Comment by samdivaio** [comment]
Get link is now removed.

File: src/NadekoBot/Modules/Music/Classes/Song.cs (hunk 1 above)
Comment (Kwoth) [inline src/NadekoBot/Modules/Music/Classes/Song.cs]: just make a single if then, checking if it's local. Radio being linked is fine.

File: src/NadekoBot/Modules/Music/Classes/Song.cs (hunk 2 above)
Comment (Kwoth) [inline src/NadekoBot/Modules/Music/Classes/Song.cs]: Not sure what you mean :s All soundcloud songs should have MusicType.SoundCloud

**Pull request merged**